### DISCUSSION TITLE
How should the notification system interact with the WordPress mobile apps.

### DISCUSSION THREAD

**Discussion by jonathanbossenger** [WordPress/wp-feature-notifications#120] [General]
Investigate how the this interacts with the WordPress Mobile apps.

We need to be able to determine how to handle the native apps, specifically:
1. how to map each notification to the WordPress mobile apps
2. more importantly: if the API should specify if a notification is only web or should show up on mobile too

Currently this is done via the WordPress XML-RPC endpoint
This might also be possible via the REST API, but Auth is a current blocker

**Comment by daniloercoli** [comment]
Right, there is still a missing piece to make the REST API accessible to mobile apps: a [central authentication scheme](https://developer.wordpress.org/rest-api/using-the-rest-api/authentication/).

This means that everything wp-notify exposes to clients (web & mobile) via the REST API, need to be added to the XML-RPC endpoint. Technically it's not a big effort, we just need to add new XML-RPC methods and put the logic in common between REST API and XML-RPC.
